homework/unique_ptr/UniquePointer.ut.cpp

```diff
@@ -0,0 +1,264 @@
+#define CATCH_CONFIG_MAIN
+#include "TestingClass.hpp"
+#include "UniquePointer.hpp"
+
+#include "catch.hpp"
+
+#include <cstddef>
+
+SCENARIO("Using unique pointers", "[uniquePtr]")
+{
+    const int firstNumber{5};
+
+    const int secondNumber{77};
+
+    std::string firstText{"Lololo"};
+
+    std::string secondText{"Bitwa pod Grunwaldem"};
+
+    GIVEN("Unique pointer to type int")
+    {
+        UniquePointer<int> ptr1(new int{firstNumber});
+
+        WHEN("Using operator*")
+        {
+            THEN("Should return value")
+            {
+                REQUIRE(*ptr1 == firstNumber);
+            }
+        }
+
+        WHEN("Using move constructor")
+        {
+            UniquePointer<int> anyPtr = std::move(ptr1);
+
+            THEN("Should create unique pointer using move constructor and free memory")
+            {
+                REQUIRE(ptr1.get() == nullptr);
+                REQUIRE(*anyPtr == firstNumber);
+            }
+        }
+
+        WHEN("Using move assigment operator")
+        {
+            UniquePointer<int> anyPtr;
+            anyPtr = std::move(ptr1);
```

Comment:
```suggestion
            UniquePointer<int> anyPtr{new int{someValue}};
            anyPtr = std::move(ptr1);
```

This version is better, because you may have memory leaks detected when you implementation is wrong. 
However your implementation of move assignment is correct, but with this test you are not able to detect memory leaks.